internal/consensus/reactor.go

```diff
@@ -732,6 +732,62 @@ func (conR *Reactor) gossipDataForCatchup(logger log.Logger, rs *cstypes.RoundSt
 	time.Sleep(conR.conS.config.PeerGossipSleepDuration)
 }
 
+func pickVoteToSend(
+	logger log.Logger,
+	conS *State,
+	rs *cstypes.RoundState,
+	ps *PeerState,
+	prs *cstypes.PeerRoundState,
+) *types.Vote {
+	// If height matches, then send LastCommit, Prevotes, Precommits.
+	if rs.Height == prs.Height {
```

Comment:
I wonder whether this logic could be outside, in the original call.

Reply:
Discussed in person. I like the fact that the three cases (same height, off-by-1, and off-by-2-or-more) are located at the same place. This was the case before this refactoring, so I'd leave it as it is